app/models/hold.rb

```diff
@@ -116,22 +116,7 @@ def self.extend_started_holds_until(date)
   def self.start_waiting_holds(now = Time.current, &block)
     started = 0
 
-    active(now).includes(item: :borrow_policy).find_each do |hold|
-      if hold.item.status == Item.statuses[:maintenance]
-        Rails.logger.debug "[hold #{hold.id}]: item in maintenance mode"
-        next
-      end
-
-      unless hold.item.holdable?
-        Rails.logger.debug "[hold #{hold.id}]: item is not holdable"
-        next
-      end
-
-      if hold.started?
-        Rails.logger.debug "[hold #{hold.id}]: already started"
-        next
-      end
-
```

Comment:
All of these debug lines were useful back in the beginning, but I think cleaning them up at this point is nice. Probably better test coverage is the better way to ensure that things are working as intended (thanks for adding another test).

I think pulling these checks up into the query makes sense.